### ISSUE TITLE
problematic doc type when sending campaign

### ISSUE THREAD

**Issue by manogi** [spatie/Mailcoach#51]
Hey there! So I stumbled upon something weird, which is that my test mails have a different doctype then my actual campaign mails. The reason for this is the following: I declare the following in my layout:

```
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
```

This stays when sending the test mail. But when sending the campaign, the `email_html` field is filled and used. In your class `Models\Campaign.php` you use the `TijsVerkoyen\CssToInlineStyles\CssToInlineStyles` for that, and THERE my doctype is stripped and replaced by the following:

```
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN" "http://www.w3.org/TR/REC-html40/loose.dtd">
```

Which ist not acceptable, as it does not interpret the font sizes of some web fonts and some other css declarations, and for some strange reason breaks scrolling on apple mail (!)

I would really like to create a PR, but I don't know where to start, to be honest. At the moment I just run a str_replace in your class, which is of course a hack and not acceptable:

```
public function htmlWithInlinedCss(): string
    {
        $html = $this->html;

        if ($this->hasCustomMailable()) {
            $html = $this->contentFromMailable();
        }

        $html = (new CssToInlineStyles())->convert($html ?? '');

        return str_replace('HTML 4.0 Transitional//EN" "http://www.w3.org/TR/REC-html40/loose.dtd"', 'XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"', $html);
    }
```

**Comment by manogi** [comment]
P.S. I used diffchecker.com to highlight the difference between t campaign mail (left) and test mail (right):

![image](https://user-images.githubusercontent.com/1419634/91966825-ac0dc800-ed12-11ea-829a-bc85fd277ff2.png)

**Comment by riasvdv** [comment]
Hi @manogi 

Thanks for reporting this issue, it's been fixed in **2.23.9**!

The issue was actually in the PrepareEmailHtmlAction, where we strip any doctype to see if the HTML has an `<html>` tag, now we store the original doctype and add it back after doing our checks

**Comment by manogi** [comment]
Cool cool cool thanks!
